app/Http/Controllers/Api/ChannelEpisodesController.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Suyabay\Http\Controllers\Api;
+
+use DB;
+use Validator;
+use Suyabay\Channel;
+use Suyabay\Episode;
+use Suyabay\Http\Requests;
+use League\Fractal\Manager;
+use Illuminate\Http\Request;
+use League\Fractal\Resource\Item;
+use Illuminate\Mail\Mailer as Mail;
+use League\Fractal\Resource\Collection;
+use Illuminate\Support\Facades\Response;
+use Suyabay\Http\Controllers\Controller;
+use Suyabay\Http\Repository\ChannelRepository;
+use Suyabay\Http\Transformers\ChannelEpisodesTransformer;
+
+class ChannelEpisodesController extends Controller
+{
+    protected $mail;
+    protected $fractal;
+
+    public function __construct(Manager $fractal, Mail $mail)
+    {
+        $this->fractal = $fractal;
+
+        parent::__construct($mail);
+    }
+
+    /**
+     * This method takes channel name and return all episodes podcast under it.
+     *
+     * @param $request
+     * @param $name
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function getAllEpisodes(Request $request, $name, ChannelEpisodesTransformer $channelEpisodesTransformer)
+    {
+        $episodes = null;
+
+        $perPage = $request->query('limit') ? : 10;
+
+        $channel = $this->getChannelByName($name);
+
+        if (! is_null($channel)) {
+            $episodes = $this->getEpisodesByChannnelId($channel, $perPage, $request);
+
+            $episodesWithComments = $this->formatEpisodes($episodes);
+
+            $resource = new Collection($episodesWithComments, $channelEpisodesTransformer);
+            $data = $this->fractal->createData($resource)->toArray();
+
+            return Response::json($data, 200);
+
+        }
+
+        return Response::json(['message' => 'Channel not found!'], 404);
+    }
+
+    /**
+     * This method validate channel request.
+     *
+     * @param $pageLimit
+     * @param $request
+     *
+     * @return $recordsToSkip
+     */
+    public function getRecordsToSkip($pageLimit, $request)
```

Comment:
**Code Quality - Critical**
@andela-tolotin In the bid of always making our code DRY, this is method you use in your API code alot. You will need to extract this code a different Class (App\Utilities\Utility)(maybe). This will make the method available anywhere.